packages/material-ui/src/ButtonBase/Ripple.js

```diff
@@ -40,12 +37,24 @@ function Ripple(props) {
     [classes.childPulsate]: pulsate,
   });
 
+  const handleExited = useEventCallback(onExited);
+  // Ripple is used for user feedback (e.g. click or press) so we want to apply styles with the highest priority
+  useEnhancedEffect(() => {
+    if (!inProp) {
+      // react-transition-group#onExit
+      setLeaving(true);
+
+      // react-transition-group#onExited
+      const timeoutId = setTimeout(handleExited, timeout);
+      return () => clearTimeout(timeoutId);
```

Comment:
pros & cons?

```suggestion
      return () => {
        clearTimeout(timeoutId);
      }
```

Reply:
Pros: I'm lazy
Cons: I shouldn't be lazy